stream-parent/stream-core/src/main/java/org/interledger/stream/StreamConnection.java

```diff
@@ -0,0 +1,207 @@
+package org.interledger.stream;
+
+import org.interledger.core.InterledgerAddress;
+import org.interledger.stream.crypto.SharedSecret;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.primitives.UnsignedLong;
+
+import java.time.Instant;
+import java.util.Objects;
+import java.util.StringJoiner;
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * <p>The session established between two endpoints that uses a single shared secret and multiplexes multiple streams
+ * of money and/or data.</p>
+ *
+ * <p>Any given Connection in a JVM manages a unique sequence id that can be incremented for each Stream Packet sent
+ * over the connection.</p>
+ */
+public class StreamConnection {
```

Comment:
Might consider making this implement `AutoCloseable` so that in can be used in a try-with-resources block.

Reply:
Agree. Tracking in #310.